lib/AST/ASTPrinter.cpp

```diff
@@ -3475,9 +3480,9 @@ class TypePrinter : public TypeVisitor<TypePrinter> {
   ///
   /// This is necessary whenever the inner type may not normally be represented
   /// as a 'type-simple' production in the type grammar.
-  void printWithParensIfNotSimple(Type T) {
+  void printWithParensIfNotSimple(Type T, const PrintOptions &options) {
```

Comment:
This method is using a mix of `options` (from your changes) and `Options` (e.g. `switch (Options.OpaqueReturnTypePrinting) {` below).  Is that expected, or a bug? If it's expected, this really needs a comment about what exactly the options are for.

If it's a bug and this is supposed to consistently use the new options, then my suggestion would be not to pass it in like this, but instead change the callers to construct the new `TypePrinter`, since only a couple of places actually override the options and most just pass through the existing ones.